app-antivirus/clamav/clamav-0.105.1.ebuild

```diff
@@ -0,0 +1,233 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LLVM_MAX_SLOT=13
+PYTHON_COMPAT=( python3_{8..11} )
+inherit cmake flag-o-matic llvm python-any-r1 systemd tmpfiles
+
+DESCRIPTION="Clam Anti-Virus Scanner"
+HOMEPAGE="https://www.clamav.net/"
+SRC_URI="https://www.clamav.net/downloads/production/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="doc clamonacc +clamapp jit libclamav-only milter rar selinux systemd test"
+
+REQUIRED_USE="libclamav-only? ( !clamonacc !clamapp !milter )
+	clamonacc? ( clamapp )
+	milter? ( clamapp )
+	test? ( !libclamav-only )"
+
+RESTRICT="!test? ( test )"
+
+# Require acct-{user,group}/clamav at build time so that we can set
+# the permissions on /var/lib/clamav in src_install rather than in
+# pkg_postinst; calling "chown" on the live filesystem scares me.
+CDEPEND="
+	acct-group/clamav
+	acct-user/clamav
+	app-arch/bzip2
+	dev-libs/json-c:=
+	dev-libs/libltdl
+	dev-libs/libmspack
+	dev-libs/libpcre2:=
+	dev-libs/libxml2
+	dev-libs/openssl:=
+	dev-libs/tomsfastmath:=
+	>=sys-libs/zlib-1.2.2:=
+	virtual/libiconv
+	!libclamav-only? ( net-misc/curl )
+	clamapp? ( sys-libs/ncurses:= net-misc/curl )
+	elibc_musl? ( sys-libs/fts-standalone )
+	jit? ( <sys-devel/llvm-$((${LLVM_MAX_SLOT} + 1)):= )
+	milter? ( mail-filter/libmilter:= )
+	rar? ( app-arch/unrar )
+	test? ( dev-python/pytest )
+"
+
+BDEPEND="
+	virtual/pkgconfig
+	>=virtual/rust-1.56
+	doc? ( app-doc/doxygen )
+	test? (
+		${PYTHON_DEPS}
+		$(python_gen_any_dep 'dev-python/pytest[${PYTHON_USEDEP}]')
+	)
+"
+
+DEPEND="${CDEPEND}
+	test? ( dev-libs/check )"
+
+RDEPEND="${CDEPEND}
+	selinux? ( sec-policy/selinux-clamav )"
+
+python_check_deps() {
+	python_has_version -b "dev-python/pytest[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	use jit && llvm_pkg_setup
+	use test && python-any-r1_pkg_setup
+}
+
+src_configure() {
+	use elibc_musl && append-ldflags -lfts
+	use ppc64 && append-flags -mminimal-toc
+
+	local mycmakeargs=(
+		-DDATABASE_DIRECTORY="${EPREFIX}"/var/lib/clamav
+		-DAPP_CONFIG_DIRECTORY="${EPREFIX}"/etc/clamav
+		-DENABLE_EXPERIMENTAL=OFF
+		-DENABLE_JSON_SHARED=ON
+		-DENABLE_APP=$(usex clamapp ON OFF)
+		-DENABLE_MILTER=$(usex milter ON OFF)
+		-DENABLE_CLAMONACC=$(usex clamonacc ON OFF)
+		-DCLAMAV_USER="clamav"
+		-DCLAMAV_GROUP="clamav"
+		-DLLVM_ROOT_DIR="$(get_llvm_prefix -d ${LLVM_MAX_SLOT})"
+		-DLLVM_FIND_VERSION="$(best_version sys-devel/llvm:${LLVM_MAX_SLOT})"
+		-DBYTECODE_RUNTIME=$(usex jit llvm interpreter)
```

Comment:
You dropped the comments but I think they're worth having -- both for users who get curious as well as maintainers.

Reply:
Couldn't get your patch to apply cleanly; consider this a WIP in case of SIGCAT (which I've prevented four times already since I sat back down...)